read-fonts/src/tables/glyf.rs

```diff
@@ -665,7 +667,254 @@ impl fmt::Display for ToPathError {
+        if self.is_empty() {
+            return Ok(());
+        }
+
+        if self.points_pending > 0 {
+            // Take the next point in the stream
+            // Wrapping read of flags and points
+            let ix = self.ix % self.contour.points.len();
+            self.points_pending -= 1;
+            self.ix += 1;
+            let (flag, curr) = (self.contour.flags[ix], self.contour.points[ix]);
+
+            if flag.is_off_curve_quad() {
+                // quads can have 0 or 1 buffered point. If there is one draw a quad to the implicit oncurve.
+                if self.off_curve_pending > 0 {
+                    self.quad_to(curr, OnCurve::Implicit, sink)?;
+                }
+                self.push_off_curve(ix);
+            } else if flag.is_off_curve_cubic() {
+                // If this is the third consecutive cubic off-curve there's an implicit oncurve between the last two
+                if self.off_curve_pending > 1 {
+                    self.cubic_to(curr, OnCurve::Implicit, sink)?;
+                }
+                self.push_off_curve(ix);
+            } else if flag.is_on_curve() {
+                // A real oncurve! What to draw depends on how many off curves are pending.
+                match self.off_curve_pending {
+                    2 => self.cubic_to(curr, OnCurve::Explicit, sink)?,
+                    1 => self.quad_to(curr, OnCurve::Explicit, sink)?,
+                    _ => {
+                        // only zero should match and that's a line
+                        debug_assert!(self.off_curve_pending == 0);
+                        let curr = curr.map(F26Dot6::to_f32);
+                        sink.line_to(curr.x, curr.y);
+                    }
+                }
+            }
+        } else {
+            // We weren't empty when we entered and we have no moe points so we must have pending off-curves
+            match self.off_curve_pending {
+                2 => self.cubic_to(self.first_move, OnCurve::Explicit, sink)?,
+                1 => self.quad_to(self.first_move, OnCurve::Explicit, sink)?,
+                _ => debug_assert!(
+                    false,
+                    "We're in a weird state. {} pending, {} off-curve.",
+                    self.points_pending, self.off_curve_pending
+                ),
+            }
         }
+        Ok(())
+    }
+}
+
+impl<'a> Contour<'a> {
+    fn new(
+        base_offset: usize,
+        points: &'a [Point<F26Dot6>],
+        flags: &'a [PointFlags],
+    ) -> Result<Self, ToPathError> {
+        if points.len() != flags.len() {
+            return Err(ToPathError::PointFlagMismatch {
+                num_points: points.len(),
+                num_flags: flags.len(),
+            });
+        }
+        Ok(Self {
+            base_offset,
+            points,
+            flags,
+        })
+    }
+
+    fn drain(self, sink: &mut impl Pen) -> Result<(), ToPathError> {
```

Comment:
**¡bikeshed alert!** Draining a sink is not as nice a metaphor as drawing with a pen ... How about draw(pen) ? :)

Reply:
I was trying to keep the sink naming @dfrg  had going alive. I tend to agree that draw(pen) better describes what's happening, will fix.

Reply:
Done